### PR TITLE
Checkout frontend

### PR DESCRIPTION
**Pull request by janiseliang**
Adds ./checkout page to site. Front end for #15.

Also added a `h-0` class to the parent div in `components/Layout.js`. Adding an explicit height to the container allows child components to use `h-full` (height: 100%) correctly. This _should_ not cause any negative effects on other pages.
Based on suggestion here: https://stackoverflow.com/a/21836870/12132573

### CODE REVIEW

**Comment by samuelwshen** [comment]
Looked really good when I saw it! Can you post a screenshot of the page?

File: pages/checkout.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      
+      e.target.barcode.value = null;
+      e.target.quantity.value = null;
+    } else {
+      if (!barcode) {
+        this.showError("Please enter a barcode in the field to the left.")
+      } else {
+        this.showError(`Invalid Barcode (${barcode})`)  
+      }
+      // console.log(`Invalid barcode: ${barcode}`)
```
Comment (jschnapper) [inline pages/checkout.js:136]: delete if not needed

File: pages/checkout.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      this.showError("Data corruption: please retry or refresh page", 20000)
+      return
+    }
+
+    /* 
+    // deletes row entry if quantity decreased to 0
+    if (itemData[1] == 1) {
+        this.deleteItem(barcode);
+        return;
+    }
```
Comment (jschnapper) [inline pages/checkout.js:87]: please delete if not needed

**Review by jschnapper — CHANGES_REQUESTED** [review]
Clean and nice use of comments 👍 good find on the layout update -- just remove some of those commented out pieces of code and i think you're good to merge

**Comment by janiseliang** [comment]
> Looked really good when I saw it! Can you post a screenshot of the page?

image
image

**Review by jschnapper — APPROVED** [review]

**Pull request merged**